src/java.base/share/classes/java/lang/foreign/SymbolLookup.java

```diff
@@ -144,9 +144,39 @@ public interface SymbolLookup {
      * @param name the symbol name
      * @return a zero-length memory segment whose address indicates the address of
      *         the symbol, if found
+     * @see #findOrThrow(String)
      */
     Optional<MemorySegment> find(String name);
 
+    /**
+     * {@return the address of the symbol with the provided {@code name} or throws an
+     *          {@linkplain IllegalArgumentException} if no such address can be found}
+     *<p>
+     * This is a convenience method that provides better exception messages compared
```

Comment:
I would reframe this as: "this is equivalent to the following code, but with better exception message"